### ISSUE TITLE
Issue with video upscaling on RTX 4090 & a dependency question

### ISSUE THREAD

**Issue by zelenooki87** [zhengchen1999/DOVE#4]
Hello,

First, thank you for your excellent work and for open-sourcing this project.

I'm looking for recommended parameters for upscaling 480p+ videos on an RTX 4090.

However, I'm running into an issue. Even with a powerful card like the RTX 4090, I'm unable to process a video with a very small input resolution of 176x144. The process hangs indefinitely and never completes. I've tried lowering the upscaling factor, but that didn't solve the problem.

I suspect the issue might be related to a missing dependency. I was unable to install SwissArmyTransformer due to an incompatibility with my CUDA and PyTorch versions. Could this missing dependency be the reason the process is failing?

My final question is: is this project intended to run on consumer hardware? My RTX 4090 should be more than capable, so I believe the problem is likely with my setup rather than the hardware itself.

Any help would be greatly appreciated.

python inference_script.py     --input_dir datasets/demo  --model_path pretrained_models/DOVE   --output_path results/DOVE/demo --is_vae_st  --png_save  --save_format yuv444p
Loading pipeline components...:   0%|                                                                                                                                                                                                 | 0/5 [00:00<?, ?it/s]Special tokens have been added in the vocabulary, make sure the associated word embeddings are fine-tuned or trained.
Loading checkpoint shards: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 4/4 [00:03<00:00,  1.16it/s]
Loading checkpoint shards: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 5/5 [00:05<00:00,  1.17s/it]
Loading pipeline components...: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 5/5 [00:10<00:00,  2.02s/it]
Processing videos:   0%|                                                                                                                                                                                                              | 0/1 [00:00<?, ?it/s]
(dove) C:\Users\Miki\DOVE>python inference_script.py     --input_dir datasets/demo  --model_path pretrained_models/DOVE   --output_path results/DOVE/demo --is_vae_st  --png_save  --save_format yuv444p
Loading checkpoint shards: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 5/5 [00:06<00:00,  1.25s/it]
Loading pipeline components...:  20%|█████████████████████████████████████                                                                                                                                                    | 1/5 [00:06<00:26,  6.50s/it]Special tokens have been added in the vocabulary, make sure the associated word embeddings are fine-tuned or trained.
Loading checkpoint shards: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 4/4 [00:06<00:00,  1.66s/it]
Loading pipeline components...: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 5/5 [00:13<00:00,  2.73s/it]
Processing videos:   0%|                                                                                                                                                                                                              | 0/1 [00:00<?, ?it/s]Process video: PRINCRIMA.mp4 | Prompt:  | Frame: 2145 (ori: 2145; pad: 0) | Target Resolution: 240, 320 (ori: 960, 1280; pad: 0, 0) | Chunk Num: 1

**Comment by zhengchen1999** [comment]
`Processing videos: 0%| | 0/1 [00:00<?, ?it/s]Process video: PRINCRIMA.mp4 | Prompt: | Frame: 2145 (ori: 2145; pad: 0) | Target Resolution: 240, 320 (ori: 960, 1280; pad: 0, 0) | Chunk Num: 1`

Based on the logs, the frame index reaches 2145, which exceeds the recommended range for processing. While the model does not have a strict frame limit, we do not recommend processing videos with more than 100 frames at once. If you need to handle longer videos, please consider using the --chunk_len XXX option to process them in chunks.

In addition, the target resolution is (240, 320), while the original (ori) resolution is (960, 1280) and the padding is (0, 0). Theoretically, when pad=0, the target resolution should match the original. Did you modify the code? Or are you using an outdated version?

Lastly, the RTX 4090 GPU should be able to run the code normally. If you encounter out-of-memory issues, you can reduce the video resolution or enable chunk-wise processing to resolve them.

**Comment by zelenooki87** [comment]
Thanks for the reply.

In the meantime, I've upgraded to a more powerful GPU (an RTX 5090 with 32GB of VRAM). Could you recommend the optimal parameters for 480p-720p video files longer than 5 seconds (over 150 frames)? Specifically, I'm looking for values for tile_size_hw, overlap_hw, chunk_len, and overlap_t.

On that note, is it even possible to run this larger model on a consumer-grade GPU? Or should I wait for you to release a less demanding version?

Thanks!

**Comment by zhengchen1999** [comment]
Hi, we recommend setting `tile_size_hw` < 1360×768, `overlap_hw` = 32, `chunk_len` < 100, and `overlap_t` = 8.  
We haven’t extensively tested all combinations, but in general, it's better to keep each chunk reasonably large and avoid overly small partitions.

Additionally, we plan to release a more lightweight version of the model in the future.  
However, compared to existing diffusion-based VSR methods, our current approach is already efficient in both computational cost and speed.

**Comment by sherlockma11** [comment]
Why I can not run on 3090,
ori Video 256*256, 16fps
overlap_hw：16，16
overlap_t 1
chunk_len 100
tile_size_hw 128 128
